### ISSUE TITLE
Allow arbitrary length generation without GPU memory error

### ISSUE THREAD

**Issue by maxrmorrison** [maxrmorrison/torchcrepe#3]
Currently, GPU memory errors prohibit long-form pitch tracking. To fix this, the batch size should be set either by the user or automatically by querying GPU capacity (e.g., with GPUtil). If the file to process exceeds one batch in size, it will have to be detached from the compute graph. Additionally, the option to detach from the compute graph should be provided. If this is given, `torch.no_grad()` can be safely used to further reduce memory consumption.

**Comment by maxrmorrison** [comment]
Added a batch_size argument to control GPU memory utilization
